bluehoover/clickhouse.py

```diff
@@ -474,15 +484,70 @@ async def insert_batch(self, batch: list[tuple]) -> None:
             self.client = None  # Force reconnect on next attempt
             raise
 
+class FilesystemCheckpoint:
```

Comment:
I'd consider defining an abstract base class, like:

```
from abc import ABC, abstractmethod

class CheckpointManager(ABC):
    @abstractmethod
    async def get_checkpoint(self) -> int | None:
        ...

    @abstractmethod
    async def save_checkpoint(self) -> None
        ...
```

We can then have these various implementations extend our `CheckpointManager` with a consistent interface.